### ISSUE TITLE
Add function to get last day of month

### ISSUE THREAD

**Issue by electrum** [prestodb/presto#8477]
This possible to do today, but it's a bit involved:
```
presto> select date_add('day', -1, date_add('month', 1, date_trunc('month', date '2017-04-15')));
   _col0              
------------
 2017-04-30 
(1 row)
```
Oracle and Hive both have a `last_day` function.

**Comment by sdpatil** [comment]
I am working on this issue. I am new to open source so not sure what is the procedure that I need to follow to get this issue assigned to me.

Sunil

**Comment by haozhun** [comment]
Removing the "beginner-task" label because the syntax and semantics of a proposed function hasn't been determined yet.

**Comment by haozhun** [comment]
postgres/Oracle approach: https://wiki.postgresql.org/wiki/Date_LastDay

**Comment by OmerJog** [comment]
Is there an update on this one?
It's very needed

**Comment by Nihhaar** [comment]
SELECT date_trunc('month', timestamp '2017-01-04 11:48:18') + interval '1' month - interval '1' day;

**Comment by RickyMcMaster** [comment]
@Nihhaar 's solution is the best way I know of, but it would be nice for this to be added.

**Comment by aweisberg** [comment]
@electrum Seems you accidentally closed the wrong issue :-)

**Comment by electrum** [comment]
@aweisberg No, I intended to close this, as I filed it originally but no longer need it.

**Comment by ericyuliu** [comment]
This is the original commit from Trino, which added 3 functions for "last_day_of_month": https://github.com/trinodb/trino/commit/2718f1a9cb3eec1fef62e5571f966c7f7c1bdcea 

Then, in the next year, 2 of these functions were removed in this commit:
https://github.com/trinodb/trino/commit/2718f1a9cb3eec1fef62e5571f966c7f7c1bdcea
Same as the latest revision on this file, there is only 1 function being kept ```lastDayOfMonthFromDate```

Hi @aweisberg, for this cherry-pick,  should I also only keep this 1 function?   Thanks!

**Comment by ericyuliu** [comment]
This is the original commit from Trino, which added 3 functions for "last_day_of_month": https://github.com/trinodb/trino/commit/2718f1a9cb3eec1fef62e5571f966c7f7c1bdcea

Then, in the next year, 2 of these functions are starting to be refactored and moved to separate files:
https://github.com/trinodb/trino/commit/21e3ddf55a641e1c17b7e4a17689223ff1120b42
presto-main/src/main/java/io/prestosql/operator/scalar/timestamp/LastDayOfMonth.java

https://github.com/trinodb/trino/commit/62cf1510d7cec2cd762711b6b3358d7e3259ac65
presto-main/src/main/java/io/prestosql/operator/scalar/timestamptz/LastDayOfMonth.java

There is only 1 function ```lastDayOfMonthFromDate``` being kept in presto-main/src/main/java/io/prestosql/operator/scalar/DateTimeFunctions.java

Hi @aweisberg, for this task, do we want to cherry-pick only ```lastDayOfMonthFromDate``` or also the other two: ```lastDayOfMonthFromTimestamp``` and ```lastDayOfMonthFromTimestampWithTimeZone```?

Thanks!

**Comment by aweisberg** [comment]
The latter two commits are pretty hefty. I would start with just getting the basic functionality.

Some of those other commits are quite involved like variable precision timestamp with TZ. That is some high value functionality, but would also be a sizable cherry pick.
